servers/lib/server/handlers/invitetoteambycsvAnalyze.coffee

```diff
@@ -0,0 +1,101 @@
+os = require 'os'
+fs = require 'fs'
+hat = require 'hat'
+path = require 'path'
+Busboy = require 'busboy'
+parser = require 'csv-parse'
+{ generateFakeClient } = require './../client'
+
+module.exports.parserOpts = csvParseOpts =
+  delimiter: ','
+  columns: ['email', 'firstName', 'lastName', 'role']
+  relax: no # Preserve quotes inside unquoted field.
+  relax_column_count: yes
+  skip_empty_lines: true
+  trim: true # remove whitespaces
+  auto_parse: no # convert integers to numbers etc
+
+module.exports.handler = (req, rres) ->
+  generateFakeClient req, rres, (err, client, session) ->
+    return rres.status(500).send err  if err
+
+    fileName = path.join(os.tmpDir(), "team-upload-#{session.groupName}-#{hat(32)}.csv")
+
+    respond = (code, message) ->
+      fs.unlink(fileName) # clean up after yourself
+      return rres.status(code).send message
+
+
+    busboy = new Busboy {
+      headers: req.headers
+      fileSize: 1024 * 1024 # 1 mb
+    }
+
+    busboy.on 'file', (fieldname, file, filename, encoding, mimetype) ->
+      file.pipe(fs.createWriteStream(fileName))
+
+    busboy.on 'finish', ->
+      analyzeInvitations fileName, client, (err, result) ->
+        return respond 500, err if err
+
+        respond 200, result
+
+    return req.pipe busboy
+
+analyzeInvitations = (fileName, client, callback) ->
+  fs.readFile fileName, 'utf8', (err, content) ->
+    return callback err, null if err
+
+    parser content.toString('utf8'), csvParseOpts, (err, data) ->
+      return callback err, null if err
+
+      { connection: { delegate: account } } = client
+      { _id } = account
+      myEmail = null
+      account.fetchEmail (err, email) ->
+
+        myEmail = email
+
+        { JGroup } = (require './../bongo').models
+        { group: slug } = client.context
+        JGroup.one { slug }, (err, group) ->
+          return callback err, null  if err
+
+          group.fetchMembersWithEmail client, {}, (err, users) ->
```

Comment:
it is not about file size it is about team's size
On Fri, 5 Aug 2016 at 21:32, Hakan Karadis <notifications@github.com> wrote:

> In servers/lib/server/handlers/invitetoteambycsvAnalyze.coffee
> <https://github.com/koding/koding/pull/8676#discussion_r73739475>:
>
> > +    parser content.toString('utf8'), csvParseOpts, (err, data) ->
> > +      return callback err, null if err
> > +
> > +      { connection: { delegate: account } } = client
> > +      { _id } = account
> > +      myEmail = null
> > +      account.fetchEmail (err, email) ->
> > +
> > +        myEmail = email
> > +
> > +        { JGroup } = (require './../bongo').models
> > +        { group: slug } = client.context
> > +        JGroup.one { slug }, (err, group) ->
> > +          return callback err, null  if err
> > +
> > +          group.fetchMembersWithEmail client, {}, (err, users) ->
>
> We can check file size before sending?
>
> —
> You are receiving this because you were mentioned.
>
> Reply to this email directly, view it on GitHub
> <https://github.com/koding/koding/pull/8676/files/3cff649744a3b9362eb1f11f84fad7c865415778#r73739475>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AA1sLPbn3VI73SiTlGS_mUAy6V9yq1-gks5qc4HOgaJpZM4JaJep>
> .
>
-- 
Cihangir.
Software Engineer @ koding.com